src/OrleansRuntime/GrainTypeManager/TypeManager.cs

```diff
@@ -1,33 +1,34 @@
 using System;
 using System.Threading.Tasks;
-using Orleans.Runtime.Providers;
+using Orleans.Streams;
 
 namespace Orleans.Runtime
 {
     internal class TypeManager : SystemTarget, ITypeManager
     {
         private readonly GrainTypeManager grainTypeManager;
+        private readonly ImplicitStreamSubscriberTable implicitStreamSubscriberTable;
 
-        internal TypeManager(SiloAddress myAddr, GrainTypeManager grainTypeManager)
+        internal TypeManager(
+            SiloAddress myAddr,
+            GrainTypeManager grainTypeManager,
+            ImplicitStreamSubscriberTable implicitStreamSubscriberTable)
             : base(Constants.TypeManagerId, myAddr)
         {
+            if (implicitStreamSubscriberTable == null) throw new ArgumentNullException(nameof(implicitStreamSubscriberTable));
```

Comment:
In the latest revision, you'll see that this aligns to what Ben is doing since his heterogenous silos PR